src/containers/CodeContainer.jsx

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect } from 'react';
+import styled from 'styled-components';
+import { connect } from 'react-redux';
+import * as actions from '../actions/actionCreators';
+
+const Wrapper = styled.div`
+  border: 1px solid black;
+  width: 1150px;
+  height: 250px;
+  margin: 10px;
+  padding: 10px;
+  font-size: 30px;
+`;
+
+const mapStateToProps = store => ({
+  codeChallenge: store.games.codeChallenge,
+  charIndex: store.games.charIndex,
+});
+
+const mapDispatchToProps = dispatch => ({
+  moveChar: num => dispatch(actions.moveChar(num)),
+});
+
+const CodeContainer = ({ codeChallenge, charIndex, moveChar }) => {
+
+  const handleKeyPress = (e) => {
+    if (e.which === 32) e.preventDefault();
+    const keyPressChar = String.fromCharCode(e.which);
+    const currentLetter = document.querySelector('#currentLetter');
+    currentLetter.style.backgroundColor = 'yellow';
+
+    if (codeChallenge.substring(charIndex, charIndex + 5) === '<br/>' && e.which === 13) moveChar(5); // Enter
+    else if (codeChallenge.substring(charIndex, charIndex + 6) === '&nbsp;' && e.which === 32) moveChar(6); // Space
+    else if (keyPressChar === codeChallenge[charIndex]) moveChar(1); // Correct char
+    else currentLetter.style.backgroundColor = 'red';
+  };
+
+  const handleTabPress = (e) => {
+    if (e.keyCode === 9) {
+      e.preventDefault();
+      if (codeChallenge.substring(charIndex, charIndex + 12) === '&nbsp;&nbsp;') moveChar(12); // Tab
+    }
+  };
+
+  useEffect(() => {
+    const correct = document.querySelector('#correct');
```

Comment:
Great job on this logic! can we utilize state and React virtual Dom to manipulate the letters ?